### DISCUSSION TITLE
posts/imdb-film-summary-as-a-mpeg-2-video/

### DISCUSSION THREAD

**Discussion by flexiondotorg** [flexiondotorg/flexiondotorg.github.io#96] [Blog Comments]
IMDB Film Summary as a MPEG-2 video
        
[https://wimpress.com/posts/imdb-film-summary-as-a-mpeg-2-video/](https://wimpress.com/posts/imdb-film-summary-as-a-mpeg-2-video/)

**Comment by flexiondotorg** [comment]
```header
Originally posted by Chris on 2009-04-24 16:45
```

That is a pretty cool idea :)

**Comment by flexiondotorg** [comment]
```header
Originally posted by Martin Wimpress on 2009-04-24 19:22
```

Thanks Chris, I'm expecting that you will send me some improvements pretty soon ;-)

**Comment by flexiondotorg** [comment]
```header
Originally posted by Eric on 2009-06-17 20:50
```

This is very nice. I've made a few changes, such as encoding the About vid into h264 directly via ffmpeg (so the ps3 can play it natively) and confirmation prompt so i could semi-automate having it go through my whole library with a bash while loop... because "Bookies" != "Cookie's Fortune" silly IMDB... :)

I also dropped the text size and set an offset position because it was HUUUUGE on my 46" tv. Seriously, the text jumped off the screen, beat me up, and stole my wallet!

Oh... and there's a bug in texttoimage as well. The parameters are wrong on the call to imagettftext()

Anyway, I'll clean it up a bit more and send it back to ya. And to think i was getting ready to do this symlink tree manually..

**Comment by flexiondotorg** [comment]
```header
Originally posted by Eric on 2009-06-18 03:07
```

You can delete this comment... i couldn't find an email address for you.

I ended up rewriting the whole thing today:

[https://www.ypass.net/downlo...](https://www.ypass.net/downloads/IMDB-to-MPEG/IMDB-to-MPEG-v2.0.tar.gz)

You'll need to add back in your mpeg2 creation stuff. I have no clue what you're doing there :).

Look for encodeVideoM4V() and just create a encodeVideoMPEG2() or whatever, then modify the switch statement below it.

**Comment by flexiondotorg** [comment]
```header
Originally posted by Martin Wimpress on 2009-06-27 14:14
```

@Eric: Thanks very much for the improvements you have made. I'll review your code and quite possibly release a revised version in the next week or so :-)

**Comment by flexiondotorg** [comment]
```header
Originally posted by Martin Wimpress on 2009-09-20 13:50
```

I've finally got round to working with Eric and incorporating all his contributions and improvements. v2.0 of IMDB-to-MPEG is now available, find out more at the URL below.

[https://wimpress.com/ind...](https://wimpress.com/index.php/2009/09/20/imdb-film-summary-as-a-mpeg2-video-part-2/)

**Comment by flexiondotorg** [comment]
```header
Originally posted by Jukka on 2009-11-24 18:15
```

Great script! Unfortunately it seems that I am the only one who is not able create the summary file (mp4 or m2v) on my (k)ubuntu 9.10 machine based on your script. Even if I am using debugging mode I don't get any errors. I believe it is related to ffmpeg part of the script so I need to study it a bit more. Do you have any hint where to look?

**Comment by flexiondotorg** [comment]
```header
Originally posted by Martin Wimpress on 2009-12-03 16:54
```

@Jukka, I have seen this before. The encode is failing because h.264 video encoding libraries are not available to ffmpeg. On 9.04 this can be fixed by install the 'libav\*-unstripped' libraries on 9.10 I think they are called libav\*-extras'. Try the following...

sudo aptitude install ffmpeg libavcodec-extra-52

If that doesn't work you can try two things. Pass the MPEG-2 flag to the script to create MPEG-2 video rather than MPEG-4, use '-o m2v'. See the help for more details.

Or a more involve work around is to modify the script to use XviD rather than h.264. Find the ffmpeg line ~813. Find '-f mp4' and change it to '-f libxvid'.

**Comment by flexiondotorg** [comment]
```header
Originally posted by Jukka on 2010-01-14 19:09
```

@Martin: Just to let you know I compiled ffmpeg from source with x264 enabled. When I used this script with m4v option, the script failed due to ffmpeg presets (I don't remember the exact error message) and no summary file was created.

Then I changed x264 to -libxvid after which the summary movie file can be created by using the script as planned. For some reason I always get the error message that movie file is not PS3 compatible. Hopefully I have some time next week to have a more closer look on this. I will let you know if I find a way to fix this (in case someone else ran into same kind of problems).

**Comment by yeshenyong** [comment]
您好，我是叶燊泳，已收到你的邮件。
